src/main/java/com/stripe/net/LiveStripeResponseGetter.java

```diff
@@ -77,6 +77,7 @@ public <T extends StripeObjectInterface> T request(
     if (resource instanceof StripeCollectionInterface<?>) {
       ((StripeCollectionInterface<?>) resource).setRequestOptions(options);
       ((StripeCollectionInterface<?>) resource).setRequestParams(params);
+      ((StripeCollectionInterface<?>) resource).setTypeToken(typeToken);
```

Comment:
Will this work for nested collections?

Reply:
Not 100% clear what you mean. If you mean
```java
var stripe = StripeClient(...);
FooResource foo = stripe.foos().retrieve(...);
for (Bar bar: foo.getNestedCollectionOfBars().autoPagingIter()) {
  System.out.println(bar);
}
```
Then it won't work but is not needed atm because the way resources are generated right now, the type of `.getNestedCollectionOfBars()` is `BarCollection`, not `StripeCollection<Bar>`.

Reply:
Ah I see the test failure, yeah there's something that needs to be done here.

Reply:
Ok, I added  [Fall back to class](https://github.com/stripe/stripe-java/pull/1663/commits/518c39414ed04619c05840421775d8cd8133dfb6) -- so that  `foo.getBars().autoPaginatingIter()` will continue to work as before.

This seemed like the minimal option but there are others:
* Generate specialized collection types as we do for resources and give up on being generic.
* Do a lookup into `EventDataClassLookup` of `page[0].object` and do dynamic class lookup instead of static class lookup.

I think I prefer what's implemented but LMK if you have other thoughts.

Reply:
I was thinking about custom type adapter for xxxCollection and xxxSearchResult that will set this field for all objects that are deserialized.

Reply:
Using what information? `page[0].object`?

Reply:
You have the type toke in the deserializer: https://github.com/stripe/stripe-java/blob/master/src/main/java/com/stripe/model/BalanceTransactionSourceTypeAdapterFactory.java#L22

Reply:
Done!